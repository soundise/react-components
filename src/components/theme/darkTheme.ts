import type { ITheme } from './theme';
import * as typography from './typography';
import * as colors from './colors';
import { breakpoints } from './breakpoints';

export const darkTheme: ITheme = {
  typography: {
    ...typography,
  },
  colors: {
    background: colors.neutral.black,
    foreground: colors.neutral.white,
    textPrimary: colors.neutral.white,
    textSecondary: colors.neutral.black,
    textTertiary: colors.primary.pr500,
    ...colors,
  },
  breakpoints: {
    ...breakpoints,
  },
};
