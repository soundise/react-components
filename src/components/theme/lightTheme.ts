import type { ITheme } from './theme';
import * as colors from './colors';

export const lightTheme: ITheme = {
  colors: {
    background: colors.neutral.white,
    foreground: colors.neutral.black,
    ...colors,
  },
};
