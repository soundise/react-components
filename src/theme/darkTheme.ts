import type { ITheme } from './theme';
import { fonts, weights } from './typography';
import { breakpoints } from './breakpoints';
import {
  neutral,
  primary,
  gray,
  greenDark,
  yellowDark,
  redDark,
} from './colors';

export const darkTheme: ITheme = {
  name: 'Soundise Dark Theme',
  variant: 'dark',
  typography: {
    fonts,
    weights,
  },
  breakpoints,
  colors: {
    background: neutral.black,
    foreground: gray.gr900,
    border: gray.gr800,
    text: {
      primary: neutral.white,
      secondary: neutral.black,
      tertiary: primary.pr500,
    },
    status: {
      success: greenDark.g500,
      warning: yellowDark.y500,
      error: redDark.r500,
    },
    brand: {
      primary: neutral.black,
      secondary: neutral.white,
      tertiary: primary.pr500,
    },
    neutral,
    primary,
    gray,
    green: greenDark,
    yellow: yellowDark,
    red: redDark,
  },
};
