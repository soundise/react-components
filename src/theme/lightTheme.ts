import type { ITheme } from './theme';
import * as typography from './typography';
import * as colors from './colors';
import { breakpoints } from './breakpoints';

export const lightTheme: ITheme = {
  variant: 'light',
  typography: {
    ...typography,
  },
  colors: {
    background: colors.neutral.white,
    foreground: colors.neutral.black,
    textPrimary: colors.neutral.black,
    textSecondary: colors.neutral.white,
    textTertiary: colors.primary.pr500,
    neutral: colors.neutral,
    primary: colors.primary,
    gray: colors.gray,
    green: colors.greenLight,
    yellow: colors.yellowLight,
    red: colors.redLight,
  },
  breakpoints: {
    ...breakpoints,
  },
};
