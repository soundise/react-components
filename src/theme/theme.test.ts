import { lightTheme } from './lightTheme';
import { darkTheme } from './darkTheme';

describe('Theme', () => {
  it('should match light theme snapshot', () => {
    expect(lightTheme).toMatchSnapshot();
  });

  it('should match dark theme snapshot', () => {
    expect(darkTheme).toMatchSnapshot();
  });
});
