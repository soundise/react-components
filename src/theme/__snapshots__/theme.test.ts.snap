// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme snapshot 1`] = `
{
  "breakpoints": {
    "desktopLarge": "1440px",
    "desktopSmall": "1024px",
    "mobileLandscape": "480px",
    "tablet": "768px",
  },
  "colors": {
    "background": "#000000",
    "border": "#2C2C2C",
    "brand": {
      "primary": "#000000",
      "secondary": "#FFFFFF",
      "tertiary": "#8D71FF",
    },
    "foreground": "#1C1C1C",
    "gray": {
      "gr050": "#F2F2F2",
      "gr100": "#E5E5E5",
      "gr200": "#C7C7C7",
      "gr300": "#AEAEAE",
      "gr400": "#8E8E8E",
      "gr500": "#636363",
      "gr600": "#484848",
      "gr700": "#3A3A3A",
      "gr800": "#2C2C2C",
      "gr900": "#1C1C1C",
    },
    "green": {
      "g050": "#EBFFF0",
      "g100": "#D6F6DE",
      "g200": "#ACEDBC",
      "g300": "#83E39B",
      "g400": "#59DA79",
      "g500": "#30D158",
      "g600": "#26A746",
      "g700": "#1D7D35",
      "g800": "#135423",
      "g900": "#0A2A12",
    },
    "neutral": {
      "black": "#000000",
      "white": "#FFFFFF",
    },
    "primary": {
      "pr050": "#F3F1FE",
      "pr100": "#E8E3FF",
      "pr200": "#D1C6FF",
      "pr300": "#BBAAFF",
      "pr400": "#A48DFF",
      "pr500": "#8D71FF",
      "pr600": "#715ACC",
      "pr700": "#554499",
      "pr800": "#382D66",
      "pr900": "#1C1733",
    },
    "red": {
      "r050": "#FFF1F0",
      "r100": "#FFDAD8",
      "r200": "#FFB5B0",
      "r300": "#FF8F89",
      "r400": "#FF6A61",
      "r500": "#FF453A",
      "r600": "#CC372E",
      "r700": "#992923",
      "r800": "#661C17",
      "r900": "#330E0C",
    },
    "status": {
      "error": "#FF453A",
      "success": "#30D158",
      "warning": "#FFD60A",
    },
    "text": {
      "primary": "#FFFFFF",
      "secondary": "#000000",
      "tertiary": "#8D71FF",
    },
    "yellow": {
      "y050": "#FFFDF0",
      "y100": "#FFF7CE",
      "y200": "#FFEF9D",
      "y300": "#FFE66C",
      "y400": "#FFDE3B",
      "y500": "#FFD60A",
      "y600": "#CCAB08",
      "y700": "#998006",
      "y800": "#665604",
      "y900": "#332B02",
    },
  },
  "name": "Soundise Dark Theme",
  "typography": {
    "fonts": {
      "code": {
        "fallback": "monospace",
        "primary": "JetBrains Mono",
        "secondary": "ui-monospace",
      },
      "fallback": "sans-serif",
      "primary": "Inter",
      "secondary": "system-ui",
    },
    "weights": {
      "bold": 700,
      "regular": 400,
    },
  },
  "variant": "dark",
}
`;

exports[`Theme snapshot 2`] = `
{
  "breakpoints": {
    "desktopLarge": "1440px",
    "desktopSmall": "1024px",
    "mobileLandscape": "480px",
    "tablet": "768px",
  },
  "colors": {
    "background": "#FFFFFF",
    "border": "#E5E5E5",
    "brand": {
      "primary": "#000000",
      "secondary": "#FFFFFF",
      "tertiary": "#8D71FF",
    },
    "foreground": "#F2F2F2",
    "gray": {
      "gr050": "#F2F2F2",
      "gr100": "#E5E5E5",
      "gr200": "#C7C7C7",
      "gr300": "#AEAEAE",
      "gr400": "#8E8E8E",
      "gr500": "#636363",
      "gr600": "#484848",
      "gr700": "#3A3A3A",
      "gr800": "#2C2C2C",
      "gr900": "#1C1C1C",
    },
    "green": {
      "g050": "#EBFFF0",
      "g100": "#D6F4DE",
      "g200": "#AEE9BD",
      "g300": "#85DD9B",
      "g400": "#5DD27A",
      "g500": "#34C759",
      "g600": "#2A9F47",
      "g700": "#1F7735",
      "g800": "#155024",
      "g900": "#0A2812",
    },
    "neutral": {
      "black": "#000000",
      "white": "#FFFFFF",
    },
    "primary": {
      "pr050": "#F3F1FE",
      "pr100": "#E8E3FF",
      "pr200": "#D1C6FF",
      "pr300": "#BBAAFF",
      "pr400": "#A48DFF",
      "pr500": "#8D71FF",
      "pr600": "#715ACC",
      "pr700": "#554499",
      "pr800": "#382D66",
      "pr900": "#1C1733",
    },
    "red": {
      "r050": "#FFF1F0",
      "r100": "#FFD8D6",
      "r200": "#FFB1AC",
      "r300": "#FF8983",
      "r400": "#FF6259",
      "r500": "#FF3B30",
      "r600": "#CC2F26",
      "r700": "#99231D",
      "r800": "#661813",
      "r900": "#330C0A",
    },
    "status": {
      "error": "#FF3B30",
      "success": "#34C759",
      "warning": "#FFCC00",
    },
    "text": {
      "primary": "#000000",
      "secondary": "#FFFFFF",
      "tertiary": "#8D71FF",
    },
    "yellow": {
      "y050": "#FFFCF0",
      "y100": "#FFF5CC",
      "y200": "#FFEB99",
      "y300": "#FFE066",
      "y400": "#FFD633",
      "y500": "#FFCC00",
      "y600": "#CCA300",
      "y700": "#997A00",
      "y800": "#665200",
      "y900": "#332900",
    },
  },
  "name": "Soundise Light Theme",
  "typography": {
    "fonts": {
      "code": {
        "fallback": "monospace",
        "primary": "JetBrains Mono",
        "secondary": "ui-monospace",
      },
      "fallback": "sans-serif",
      "primary": "Inter",
      "secondary": "system-ui",
    },
    "weights": {
      "bold": 700,
      "regular": 400,
    },
  },
  "variant": "light",
}
`;
